docs/src/main/asciidoc/application-data-caching-guide.adoc

```diff
@@ -0,0 +1,474 @@
+
+Now that the cache was loaded during the previous call, try calling the same URL. This time, you should get a super fast answer with
+an `executionTimeInMs` value close to 0.
+
+Let's see what happens if we start from one day in the future using the
+`http://localhost:8080/weather?city=Raleigh&daysInFuture=1` URL. You should get an answer two seconds later since two of the
+requested days were already loaded in the cache.
+
+You can also try calling the same URL with a different city and see the cache in action again. The first call will take six
+seconds and the following ones will be answered immediately.  
+
+Congratulations! You just added the application data cache to your Quarkus application with a single line of code!
+
+Do you want to learn more about the Quarkus application data caching abilities? The following sections will show you everything
+there is to know about it.
+
+== Caching annotations
+
+Quarkus offers four method-level annotations and one argument-level annotation that can be used in a CDI managed bean to
+enable caching abilities:
+
+=== @CacheLoad
+
+When a method annotated with `@CacheLoad` is invoked, Quarkus will check in the cache whether the method has been already
+invoked for the given arguments. This check is done using a cache key computed from all the method arguments if none of them is
+annotated with `@CacheKey`, or all the arguments annotated with `@CacheKey` otherwise. If a value is found in the cache, it
+is returned and the annotated method is never actually executed. If no value is found, the annotated method is invoked and the
+returned value is stored in the cache using the computed key.
+
+This annotation can't be used on methods returning `void` and requires at least one method argument to compute the key.
+
+=== @CacheStore
+
+When a method annotated with `@CacheStore` is invoked, Quarkus will cache its return value at the end of the invocation. The
+caching is done using a cache key computed from all the method arguments if none of them is annotated with `@CacheKey`, or all
+the arguments annotated with `@CacheKey` otherwise. If a value is already present in the cache for the computed key, it is
+replaced.
+
+This annotation can't be used on methods returning `void` and requires at least one method argument to compute the
+
+=== @CacheInvalidate
+
+When a method annotated with `@CacheInvalidate` is invoked, Quarkus will try to remove an existing entry from the cache. This
+removal is done using a cache key computed from all the method arguments if none of them is annotated with `@CacheKey`, or all
+the arguments annotated with `@CacheKey` otherwise. If the key does not identify any cache entry, nothing will happen.
+
+This annotation requires at least one method argument to compute the key.
+
+[TIP]
+====
+If the `@CacheLoad`, `@CacheStore` or `@CacheInvalidate` annotations are used on a method with no parameters, a unique default
+cache key derived from the cache name will be generated and used.
```

Comment:
So, looks like we don't need at least one method argument?

Reply:
No, we don't :)